server/src/test/java/org/elasticsearch/gateway/PriorityComparatorTests.java

```diff
@@ -106,75 +147,102 @@ protected Settings getIndexSettings(Index index) {
-                            assertTrue("indexName mismatch, expected:" + currentMeta.name + " after " + prevMeta.name + " " +
-                                prevMeta.name.compareTo(currentMeta.name), prevMeta.name.compareTo(currentMeta.name) > 0);
+                IndexMetadata prevMeta = map.get(previous.getIndexName());
+                IndexMetadata currentMeta = map.get(routing.getIndexName());
+
+                if (prevMeta.isSystem() == currentMeta.isSystem()) {
+                    final int prevPriority = prevMeta.getSettings().getAsInt(IndexMetadata.SETTING_PRIORITY, -1);
+                    final int currentPriority = currentMeta.getSettings().getAsInt(IndexMetadata.SETTING_PRIORITY, -1);
+
+                    if (prevPriority == currentPriority) {
+                        final int prevCreationDate = prevMeta.getSettings().getAsInt(IndexMetadata.SETTING_CREATION_DATE, -1);
+                        final int currentCreationDate = currentMeta.getSettings().getAsInt(IndexMetadata.SETTING_CREATION_DATE, -1);
+
+                        if (prevCreationDate == currentCreationDate) {
+                            final String prevName = prevMeta.getIndex().getName();
+                            final String currentName = currentMeta.getIndex().getName();
+
+                            if (prevName.equals(currentName) == false) {
+                                assertTrue(
+                                    "indexName mismatch, expected:"
+                                        + currentName
+                                        + " after "
+                                        + prevName
+                                        + " "
+                                        + prevName.compareTo(currentName),
+                                    prevName.compareTo(currentName) > 0
+                                );
+                            }
+                        } else {
+                            assertTrue(
+                                "creationDate mismatch, expected:" + currentCreationDate + " after " + prevCreationDate,
+                                prevCreationDate > currentCreationDate
+                            );
                         }
                     } else {
-                        assertTrue("creationDate mismatch, expected:" + currentMeta.creationDate + " after " + prevMeta.creationDate,
-                            prevMeta.creationDate > currentMeta.creationDate);
+                        assertTrue(
+                            "priority mismatch, expected:" + currentPriority + " after " + prevPriority,
+                            prevPriority > currentPriority
+                        );
                     }
                 } else {
-                    assertTrue("priority mismatch, expected:" +  currentMeta.priority + " after " + prevMeta.priority,
-                        prevMeta.priority > currentMeta.priority);
+                    assertTrue(
```

Comment:
Maybe 

assertThat("system mismatch, expected:" + currentMeta.isSystem() + " after " + prevMeta.isSystem(),
Boolean.compare(prevMeta.isSystem(), currentMeta.isSystem()), greaterThanOrEqualTo(0));

Reply:
I changed it to a modified version of your suggestion, using `greaterThan()`. I also changed the other assertions for consistency.